main.tf

```diff
@@ -1,32 +1,106 @@
-# TODO: insert resources here.
 data "azurerm_resource_group" "parent" {
   count = var.location == null ? 1 : 0
-  name  = var.resource_group_name
+
+  name = var.resource_group_name
 }
 
-resource "azurerm_TODO_the_resource_for_this_module" "this" {
-  name                = var.name # calling code must supply the name
+data "azurerm_mssql_server" "this" {
+  count = var.existing_parent_resource != null ? 1 : 0
+
+  name                = var.existing_parent_resource.name
   resource_group_name = var.resource_group_name
-  location            = coalesce(var.location, local.resource_group_location)
-  # etc
+}
+
+resource "azurerm_mssql_server" "this" {
+  count = var.existing_parent_resource == null ? 1 : 0
+
+  location                                     = try(data.azurerm_resource_group.parent[0].location, var.location)
+  name                                         = var.name # calling code must supply the name
+  resource_group_name                          = var.resource_group_name
+  version                                      = var.server_version
+  administrator_login                          = var.administrator_login
+  administrator_login_password                 = var.administrator_login_password
+  connection_policy                            = var.connection_policy
+  minimum_tls_version                          = "1.2"
+  outbound_network_restriction_enabled         = var.outbound_network_restriction_enabled
+  primary_user_assigned_identity_id            = var.primary_user_assigned_identity_id
+  public_network_access_enabled                = var.public_network_access_enabled
+  tags                                         = var.tags
+  transparent_data_encryption_key_vault_key_id = var.transparent_data_encryption_key_vault_key_id
+
+  dynamic "azuread_administrator" {
+    for_each = var.azuread_administrator != null ? { this = var.azuread_administrator } : {}
+
+    content {
+      login_username              = azuread_administrator.value.login_username
+      object_id                   = azuread_administrator.value.object_id
+      azuread_authentication_only = azuread_administrator.value.azuread_authentication_only
+      tenant_id                   = azuread_administrator.value.tenant_id
```

Comment:
What happens if this is null, should we be supplying the tenant id from the Terraform config in that scenario?

Reply:
Done